cmd/main.go

```diff
@@ -1,79 +1,201 @@
 	"time"
 
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
 	pc "github.com/opiproject/opi-api/inventory/v1/gen/go"
 	pe "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
 	"github.com/opiproject/opi-evpn-bridge/pkg/bridge"
+	"github.com/opiproject/opi-evpn-bridge/pkg/config"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/taskmanager"
 	"github.com/opiproject/opi-evpn-bridge/pkg/port"
 	"github.com/opiproject/opi-evpn-bridge/pkg/svi"
 	"github.com/opiproject/opi-evpn-bridge/pkg/utils"
 	"github.com/opiproject/opi-evpn-bridge/pkg/vrf"
 	"github.com/opiproject/opi-smbios-bridge/pkg/inventory"
 
-	"github.com/philippgille/gokv"
-	"github.com/philippgille/gokv/redis"
-
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/reflection"
 
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-
+	ci_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxCIModule"
+	gen_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxGeneralModule"
+	ipu_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxVendorModule/ipu"
+	frr "github.com/opiproject/opi-evpn-bridge/pkg/frr"
 	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 )
 
-func main() {
-	var grpcPort int
-	flag.IntVar(&grpcPort, "grpc_port", 50151, "The gRPC server port")
+const (
+	configFilePath = "./"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "opi-evpn-bridge",
+	Short: "evpn bridge",
+	Long:  "evpn bridge application",
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateConfigs()
+	},
+	Run: func(_ *cobra.Command, _ []string) {
+
+		taskmanager.TaskMan.StartTaskManager()
 
-	var httpPort int
-	flag.IntVar(&httpPort, "http_port", 8082, "The HTTP server port")
+		err := infradb.NewInfraDB(config.GlobalConfig.DBAddress, config.GlobalConfig.Database)
+		if err != nil {
+			log.Println("error in creating db", err)
+		}
+		go runGatewayServer(config.GlobalConfig.GRPCPort, config.GlobalConfig.HTTPPort)
 
-	var tlsFiles string
-	flag.StringVar(&tlsFiles, "tls", "", "TLS files in server_cert:server_key:ca_cert format.")
+		defer func() {
+			if err := infradb.Close(); err != nil {
+				log.Fatal(err)
+			}
+		}()
 
-	var redisAddress string
-	flag.StringVar(&redisAddress, "redis_addr", "127.0.0.1:6379", "Redis address in ip_address:port format")
+		switch config.GlobalConfig.Buildenv {
+		case "ipu":
+			gen_linux.Init()
+			ipu_linux.Init()
+			frr.Init()
+
+		case "ci":
+			gen_linux.Init()
+			ci_linux.Init()
+			frr.Init()
+		default:
+			log.Fatal(" ERROR: Could not find Build env ")
+		}
 
-	var frrAddress string
-	flag.StringVar(&frrAddress, "frr_addr", "127.0.0.1", "Frr address in ip_address format, no port")
+		// Create GRD VRF configuration during startup
+		if err := createGrdVrf(); err != nil {
+			log.Printf("Error in creating GRD VRF %+v\n", err)
```

Comment:
No it cannot function because the GRD represents the underlay configuration which will be needed in the next drop [drop-1.0] where we will provide the fastpath modules which are responsible for offloading the slowpath configuration to the fastpath. So in order to configure correctly the fastpath for the underlay the GRD vrf is needed to be there so the faspath modules can find the correct inofrmation and offload the rules.

Reply:
so let's return an error from this function. In the current state, it will continue executing, which is not coorect

Reply:
Yes ok we wil take care of this

Reply:
added the error log and exit call if grd vrf is not created

Reply:
Done